Editor: Listen for Command-Z instead of Ctrl-Z on Mac

TyLindberg opened the pull request:
Conditionally change the modifier key to Command if the platform is detected as Mac.

Commit:
Listen for Command-Z instead of Ctrl-Z on Mac

TyLindberg commented:
With the current implementation, it checks if the platform is Mac on every press of the 'z' key. Ideally we'd only do this once, but I'm not sure where would be best to store that value.

Mugen87 commented:
I like the idea of utilizing the command key. You can maybe put the platform check at this place.

https://github.com/TyLindberg/three.js/blob/8466a3e97b857c0671ba4481eac2b4ac68eb5c72/editor/index.html#L165

Commit:
Moved platform check out of input handler

TyLindberg commented:
Ok, I just moved that platform check out of the input handler and also moved out the modifier key conditional since that should stay constant as well. A possible issue is that `modifierKey` is now bound to global, though I'm not sure how big of worry that is.

Mugen87 commented on editor/index.html:
```diff
@@ -311,11 +315,11 @@
 					case 90: // Register Ctrl-Z for Undo, Ctrl-Shift-Z for Redo
 
-						if ( event.ctrlKey && event.shiftKey ) {
+						if ( event[modifierKey] && event.shiftKey ) {
```
Just some code style issue: Use spaced between brackets
```javascript
event[ modifierKey ]
```

Mugen87 commented on editor/index.html:
```diff
@@ -163,6 +163,10 @@
 
+			// Use Command instead of Ctrl on Mac
+			var modifierKey = navigator.platform.toUpperCase().indexOf( 'MAC' ) >= 0 ?
+				"metaKey" : "ctrlKey";
```
I would write this in a single line.

Mugen87 commented:
I just thought i might be useful to have this information in global space so it's easier to use in at other places. But yes, it is not the cleanest solution. We maybe can transfer this information to a property of a prototype like `Editor`. I'm sure @mrdoob has an idea for this.

Commit:
Styling fixes

TyLindberg commented:
Yeah, sounds good to me. I think in the future it would be cool to get the input hooked up to the settings tab so that users could change their key bindings. That would maybe handle the encapsulation issue as well.

mrdoob commented:
How about this?

```js
if ( ( event.ctrlKey || event.metaKey ) && event.shiftKey ) {
```

TyLindberg commented:
Yeah, that seems pretty reasonable to me. My only concern would be that `metaKey` might capture the Windows key on Windows platform as per [this documentation](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/metaKey). However, it also seems like that may be old behavior that we don't have to worry about. Unfortunately I don't have a Windows machine to test on right now to explore that.

mrdoob commented:
Ah, I see. Wasn't aware that `metaKey` was mapped to the Windows key...

Then, lets add this to the top of the script in `index.html`.

```js
const IS_MAC = navigator.platform.toUpperCase().indexOf( 'MAC' );
```

And then do:

```js
if ( ( IS_MAC ? event.metaKey : event.ctrlKey ) && event.shiftKey ) {
```

Commit:
Restructured editor platform check

TyLindberg commented:
Cool, just implemented it in that fashion. I also restructured it so that the `IS_MAC` conditional was only checked once to avoid duplicating code.

mrdoob commented:
Even better!

mrdoob commented:
Thanks!

TyLindberg commented:
Thank you!
